test/trajectory.jl

```diff
@@ -186,29 +186,33 @@ function gettraj(rng, ϵ=0.1, n_steps=50)
     return traj_z
 end
 
-function hand_isturn(z0, z1, rho, v=1)
+function hand_isturn_classic(z0, z1, rho, v=1)
     θ0minusθ1 = z0.θ - z1.θ
     s = (dot(-θ0minusθ1, -z0.r) >= 0) || (dot(θ0minusθ1, z1.r) >= 0)
     return s
 end
 
-ahmc_isturn(z0, z1, rho, v=1) =
-    AdvancedHMC.isterminated(h, AdvancedHMC.BinaryTree(z0, z1, ClassicNoUTurn(), 0, 0, 0.0)).dynamic
+function ahmc_isturn_classic(z0, z1, rho, v=1)
+    tree = AdvancedHMC.BinaryTree(z0, z1, nothing, 0, 0, 0.0)
+    return AdvancedHMC.isterminated(ClassicNoUTurn(), h, tree).dynamic
+end
 
-function hand_isturn_generalised(z0, z1, rho, v=1)
+function hand_isturn(z0, z1, rho, v=1)
```

Comment:
It's ok to keep the current name `hand_isturn_generalised`, which is more informative.

Reply:
I just want make it clear why I did this name change of `GeneralisedNoUTurn` -> `NoUTurn`.
Originally, we only have the classic no-U-turn and when we add the generalised no-U-turn we want to make it clear that it's the generalised version.
However, I believe when refer to no-U-turn, people now by default refer to the generalised version.
So in this context, `NoUTurn` by default refers to the generalised version and we use the explicit `Classic` prefix to refer to the original version and the `Strict` prefix refers to a more strict version of `NoUTurn`.

I'm fine to revert these changes though and have some more discussion on this before commiting it.

Reply:
Yes, let's keep things descriptive internally. We can create a type alias for `const NoUTurn=GeneralisedNoUTurn`.